language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/SnippetBlock.java

```diff
@@ -71,30 +72,33 @@ public SnippetBlock(String label, String snippet, String detail, Kind kind,
      * @return modified Completion Item
      */
     public CompletionItem build(BallerinaCompletionContext ctx) {
+        // Each time we build a new completion item to make the snippet aware of the latest sources (alias changes, etc)
         CompletionItem completionItem = new CompletionItem();
-        completionItem.setInsertText(this.snippet);
-        List<ImportDeclarationNode> currentDocImports = ctx.currentDocImports();
+        String insertText = this.snippet;
         if (imports != null) {
             List<TextEdit> importTextEdits = new ArrayList<>();
             for (Pair<String, String> pair : imports) {
-                boolean pkgAlreadyImported = currentDocImports.stream()
-                        .anyMatch(importNode -> importNode.orgName().isPresent()
-                                && importNode.orgName().get().orgName().text().equals(pair.getLeft())
-                                && importNode.prefix().isPresent()
-                                && importNode.prefix().get().prefix().text().equals(pair.getRight()));
-                if (!pkgAlreadyImported) {
+                Optional<ImportDeclarationNode> matchedImport = CommonUtil.matchingImportedModule(ctx, pair.getLeft(),
+                        pair.getRight());
+                if (matchedImport.isEmpty()) {
                     importTextEdits.addAll(CommonUtil.getAutoImportTextEdits(pair.getLeft(), pair.getRight(), ctx));
                 }
+                if (matchedImport.isPresent() && matchedImport.get().prefix().isPresent()) {
```

Comment:
This can be moved to `else if` of the same if ladder.. Then we can remove redundant `isPresent()` check.

eg. `} else if (matchedImport.get().prefix().isPresent()){...}`

Reply:
Fixed